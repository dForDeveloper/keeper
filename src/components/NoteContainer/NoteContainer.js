import React from 'react'
import './NoteContainer.scss'

const NoteContainer = () => {

  return (
    <div className ='NoteContainer'></div>
  )
}

export default NoteContainer