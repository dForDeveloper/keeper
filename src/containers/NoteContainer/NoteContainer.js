import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NoteCard } from '../NoteCard/NoteCard';
import { Link } from 'react-router-dom';
import Masonry from 'react-masonry-css';
import Loader from '../../components/Loader/Loader';
import newnoteicon from '../../images/newnoteicon.svg';

export const NoteContainer = ({ notes, isLoading, isDisabled }) => {
  const cards = notes.map(note => {
    return <NoteCard {...note} key={note.id}/>
  }).reverse();

  const breakpoints = {
    default: 6,
    1552: 5,
    1296: 4,
    1040: 3,
    784: 2,
    528: 1
  };

  const disabledClass = isDisabled ? '--disabled' : '';

  return (
    <div className={'NoteContainer' + disabledClass}>
      <div className='NoteContainer--div'>
        <Link to='/new-note' className='NoteContainer--link'>
          <img
            src={newnoteicon}
            className='NoteContainer--icon--new-note'
            alt='new note icon'
          />
          <span className='NoteContainer--span'>
            New Note
          </span>
        </Link>
      </div>
      {isLoading && <Loader />}
      {!isLoading &&
        <Masonry
          breakpointCols={breakpoints}
          className='NoteContainer--cards'
          columnClassName='NoteContainer--cards-masonry-cols'>
          {cards}
        </Masonry>}
    </div>
  );
};

export const mapStateToProps = (state) => ({
  notes: state.notes,
  isLoading: state.isLoading
});

export default connect(mapStateToProps)(NoteContainer);

NoteContainer.propTypes = {
  notes: PropTypes.array,
  isLoading: PropTypes.bool,
  isDisabled: PropTypes.bool
}