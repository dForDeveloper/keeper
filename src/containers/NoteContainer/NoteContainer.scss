.NoteContainer {
  width: 1536px;
  margin: auto;
}

.NoteContainer--disabled {
  @extend .NoteContainer;
  pointer-events: none;
  filter: opacity(0.9) blur(.7px);
}

.NoteContainer--div {
  width: 100%;
  display: grid;
  margin: 24px 0 16px;
  justify-content: center;
  grid-template-columns: 48px 88px 48px;
}

.NoteContainer--link {
  display: block;
  font-size: 1.2rem;
  text-decoration: none;
  color: #000;
  display: grid;
  grid-template-columns: 48px 88px;
}

.NoteContainer--cards {
  display: flex;
  width: auto;
}

.NoteContainer--cards-masonry-cols {
  background-clip: padding-box;
}

.NoteContainer--icon--new-note {
  width: 22px;
  justify-self: right;
  margin-right: 14px;
  border-radius: 5px;
  box-shadow: $light-shadow;
  
  &:hover {
    box-shadow: $heavy-shadow;
  }
}

.NoteContainer--span {
  justify-self: left;
  align-self: center;
}

@media screen and (max-width: 1552px) {
  .NoteContainer {
    width: 1280px;
  }
}

@media screen and (max-width: 1296px) {
  .NoteContainer {
    width: 1024px;
  }
}

@media screen and (max-width: 1040px) {
  .NoteContainer {
    width: 768px;
  }
}

@media screen and (max-width: 784px) {
  .NoteContainer {
    width: 512px;
  }
}

@media screen and (max-width: 528px) {
  .NoteContainer {
    width: 100%;
  }
}