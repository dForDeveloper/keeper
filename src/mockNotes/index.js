export const mockNotes = [
  { 
    id: 'ijf', 
    title: 'Learn Redux', 
    listItems: [
      { 
        id: 'asg', 
        description: 'read through lesson', 
        isComplete: false
      }, 
      { 
        id: 'rhf', 
        description: 'follow along with tutorial', 
        isComplete: false
      }
    ]
  },
  { 
    id: 'pdo', 
    title: 'Start portfolio', 
    listItems: [
      { 
        id: 'cat', 
        description: 'Setup user account', 
        isComplete: false
      }
    ]
  },
  { 
    id: 'bah', 
    title: 'Grocery shopping', 
    listItems: [
      { 
        id: 'rey', 
        description: 'Stop by King Soopers', 
        isComplete: false
      }, 
      { 
        id: 'moo', 
        description: 'Stop by Sprouts', 
        isComplete: false
      }
    ]
  },
  { 
    id: 'fah', 
    title: 'Make dinner', 
    listItems: [
      { 
        id: 'ley', 
        description: 'gather ingredients', 
        isComplete: false
      }, 
      { 
        id: 'mei', 
        description: 'preheat oven', 
        isComplete: false
      }
    ]
  }
]

export const mockNote = {
  id: 'rzz',
  title: 'Workout',
  listItems: [
    {
      id: 'lpo',
      description: '20 situps',
      isComplete: false
    },
    {
      id: 'lqw',
      description: '15 pushups',
      isComplete: false
    }
  ]
}