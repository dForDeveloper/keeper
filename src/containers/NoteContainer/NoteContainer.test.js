import React from 'react';
import { shallow } from 'enzyme';
import { NoteContainer, mapStateToProps } from './NoteContainer';
import { mockNotes } from '../../mockNotes';

describe('NoteContainer', () => {
  it('should match the snapshot', () => {
    let wrapper = shallow(<NoteContainer notes={mockNotes} isLoading={false}/>)
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {
  it('should return props with notes and isLoading', () => {
    const mockState = {
        notes: mockNotes,
        isLoading: false,
        error: '',
        status: 0
    };
    const expected = { notes: mockNotes, isLoading: false };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});