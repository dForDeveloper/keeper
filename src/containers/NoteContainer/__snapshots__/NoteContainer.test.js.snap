// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteContainer should match the snapshot 1`] = `
<div
  className="NoteContainer"
>
  <Link
    className="NoteContainer--new-note"
    replace={false}
    to="/new-note"
  >
    New Note
  </Link>
  <Masonry
    breakpointCols={
      Object {
        "1100": 3,
        "1500": 4,
        "550": 1,
        "790": 2,
        "default": 5,
      }
    }
    className="NoteContainer--cards"
    columnClassName="NoteContainer--cards-masonry-cols"
  >
    <NoteCard
      id="fah"
      key="fah"
      listItems={
        Array [
          Object {
            "description": "gather ingredients",
            "id": "ley",
            "isComplete": false,
          },
          Object {
            "description": "preheat oven",
            "id": "mei",
            "isComplete": false,
          },
        ]
      }
      title="Make dinner"
    />
    <NoteCard
      id="bah"
      key="bah"
      listItems={
        Array [
          Object {
            "description": "Stop by King Soopers",
            "id": "rey",
            "isComplete": false,
          },
          Object {
            "description": "Stop by Sprouts",
            "id": "moo",
            "isComplete": false,
          },
        ]
      }
      title="Grocery shopping"
    />
    <NoteCard
      id="pdo"
      key="pdo"
      listItems={
        Array [
          Object {
            "description": "Setup user account",
            "id": "cat",
            "isComplete": false,
          },
        ]
      }
      title="Start portfolio"
    />
    <NoteCard
      id="ijf"
      key="ijf"
      listItems={
        Array [
          Object {
            "description": "read through lesson",
            "id": "asg",
            "isComplete": false,
          },
          Object {
            "description": "follow along with tutorial",
            "id": "rhf",
            "isComplete": false,
          },
        ]
      }
      title="Learn Redux"
    />
  </Masonry>
</div>
`;
